book changes (#28)
leudz/shipyard: book/src/concepts/running-the-world-with-systems.md

There are two ways to operate on the world.

## Systems

> The most convenient way is by using the `system` annotation and then registering it as a workload:

Reviewer: I agree, showing it without hinting there's light at the end of the tunnel might scare some people off

```suggestion
Systems are created by implementing the `System` trait. It's a bit involved- but don't worry, there's a helper macro we'll get to down below that makes this much simpler!

First, implementing the trait directly. It looks like:
```

1. Define a system
```rust, noplaypen
struct CreateEmpty;
impl<'a> System<'a> for CreateEmpty {
    type Data = (EntitiesMut, &'a mut Empty);
    
    fn run((entities, empties): <T::Data as SystemData>::View) { ... }
}
```

First we create a struct, it could be an enum, empty or not it doesn't really matter it's just here to attach the impl on. Note that even if you don't make it empty you won't have access to the struct inside the system.

> We then choose which storages we want with `Data`. We'll see in the next chapters what types can be used in detailed.

Reviewer: ```suggestion
We then choose which storages we want with the `Data` associated type. We'll see in the next chapters what types can be used in detailed.
```

> Finally we get a tuple containing views to the storages we asked with `Data`. You can use `<T::Data as SystemData>::View` as type, it's a bit esoteric but will work whatever you borrow. Or you could specify the exact types.

Reviewer: ```suggestion
Finally, we get a tuple as `run` argument containing views to the storages we asked with `Data`. You can use `<T::Data as SystemData>::View` as type, it's a bit esoteric but will work with whatever you borrow. Or you could specify the exact types.
```

2. Add the workload
```rust, noplaypen
world.add_workload("Creators", CreateEmpty);
```

3. Run the workload (which will run its systems)
```rust, noplaypen
world.run_workload("Creators");
```

Adding multiple systems to a workload is only a matter of expanding the second argument to a tuple. For example: 

> 1. Define another system

Reviewer: ```suggestion
1. Define another system - this time using the macro
```

```rust, noplaypen
#[system(CreateCount)]
fn run (entities: &mut Entities, counts: &mut Count) { ... }
```

> This time we use the macro, it'll make the struct and impl for you! And no tuple, the macro will also take care of it.

Reviewer: ```suggestion
It will also take care of transforming its arguments into tuple(s), no need to worry about it anymore.
```

2. Add the workload
```rust, noplaypen
world.add_workload("Creators", (CreateEmpty, CreateCount));
```

3. Run the workload
```rust, noplaypen
world.run_workload("Creators");
```

This is great because it avoids a ton of boilerplate and provides [outer-parallelism](../concepts/inner-outer-parallelism.md) without having to do anything.

There's a few points to keep in mind about workloads:

> 1. Workloads will run its systems in parallel where possible. If they can't be run in parallel, then systems run sequentially left-to-right.

Author: The above blocks are being moved to a new chapter, yeah? Just didn't see it in this PR ;)

Reviewer: Yes, it'll be part of the next push with the two previous chapters, I have it in local
